site/testsuites/testsuite-javascript/__tests__/tbdex/pfi/pfi-structure.js

```diff
@@ -0,0 +1,243 @@
+    // Mock data for getRfq
+    const mockData = { id: opts.exchangeId, type: 'Rfq' };
+    return Promise.resolve(mockData);
+  },
+
+  getQuote: (opts) => {
+    // Mock data for getQuote
+    const mockData = { id: opts.exchangeId, type: 'Quote' };
+    return Promise.resolve(mockData);
+  },
+
+  getOrder: (opts) => {
+    // Mock data for getOrder
+    const mockData = { id: opts.exchangeId, type: 'Order' };
+    return Promise.resolve(mockData);
+  },
+
+  getOrderStatuses: (opts) => {
+    // Mock data for getOrderStatuses
+    const mockData = [{ status: 'Pending' }, { status: 'Completed' }];
+    return Promise.resolve(mockData);
+  },
+
+  getClose: (opts) => {
+    // Mock data for getClose
+    const mockData = { id: opts.exchangeId, type: 'Close' };
+    return Promise.resolve(mockData);
+  },
+
+  // :snippet-start: pfiOverviewWriteJs
+  async write({ message }) {
+      await dataProvider.insert('exchange', {
+              exchangeid: message.exchangeId,
+              messagekind: message.kind,
+              messageid: message.id,
+              subject: message.subject,
+              message: JSON.stringify(message)
+          });
+  }  
+  // :snippet-end:
+};
+
+// :snippet-start: pfiOverviewReadOfferingsJs
+const offeringsApiProvider = {
+  getOffering: (opts) => {
+      dataProvider.get('offering', opts.id).then(([result]) => {
+          if (!result) {
+              return undefined
+          }
+          return Offering.factory(result.offering)
+      });
+  },
+
+  getOfferings: (opts) => {
+    dataProvider.query('offering', "*").then((results) => {
+        const offerings = []
+  
+        for (let result of results) {
+            const offering = Offering.factory(result.offering)
+            offerings.push(offering)
+        }
+    
+        return offerings
+    });
+  },
+
+  // :snippet-end:
+
+  // :snippet-start: pfiOverviewWriteOfferingsJs
+  async create(offering) {
+      await dataProvider.insert('offering', {
+          offeringid: offering.id,
+          payoutcurrency: offering.payoutCurrency.currencyCode,
+          payincurrency: offering.payinCurrency.currencyCode,
+          offering: JSON.stringify(offering)
+      });
+  }
+    // :snippet-end:
+};  
+
+// :snippet-start: pfiOverviewConfigJs
+const tbDexServer = new TbdexHttpServer({ 
+  exchangesApi: exchangesApiProvider, 
+  offeringsApi: offeringsApiProvider 
+})
+// :snippet-end:
+
+// :snippet-start: pfiOverviewServerRoutesJs
+tbDexServer.submit('rfq', async (ctx, rfq) => {
+    await exchangesApiProvider.write({ message: rfq})
+})
+
+tbDexServer.submit('order', async (ctx, order) => {
+    await exchangesApiProvider.write({ message: order })
+})
+
+tbDexServer.submit('close', async (ctx, close) => {
+    await exchangesApiProvider.write({ message: close })
+})
+// :snippet-end:
+
+// :snippet-start: pfiOverviewServerStartJs
+const server = tbDexServer.listen(8080, () => {
+    console.log.info(`PFI listening on port ${8080}`)
```

Comment:
```suggestion
    console.log('PFI listening on port 8080')
```

This is not valid JS. Please test code before committing it.

Reply:
Fixed